src/DefaultUnleash.php

```diff
@@ -32,7 +33,8 @@ public function __construct(
 
     public function isEnabled(string $featureName, ?Context $context = null, bool $default = false): bool
     {
-        $context ??= $this->configuration->getContextProvider()->getContext();
+        $defaultContext = $this->configuration->getContextProvider()->getContext();
+        $context = $context ? self::mergeContext($defaultContext, $context) : $defaultContext;
```

Comment:
Why is this necessary? I can't really see the point of merging with default context instead of either using custom or the default.

Reply:
Seems you're right. I just realized that we do not need it. But, the initial idea was taken from the node client. https://github.com/Unleash/unleash-client-node/blob/master/src/unleash.ts#L213
```
isEnabled(name: string, context: Context = {}, fallback?: FallbackFunction | boolean): boolean {
    const enhancedContext = { ...this.staticContext, ...context };
...
```

Reply:
The idea is that there exist a few "static" context fields (`appName` and `environment`) that does not change for a given application per request. It is thus convenient to not have to pass these for every call just because the `userId` changes. 

All other official SDKs currently follow this pattern. One could of course argue that this logic can be handled outside the SDK.

Reply:
I see, makes sense, but it's anyway out of scope of this PR IMHO.

Reply:
Agree. I removed this change from the PR and pushed fixes for tests.